modmesh/onedim/euler1d.py

```diff
@@ -302,16 +302,27 @@ def calc_internal_energy(self, pressure, density):
     def calc_entropy(self, pressure, density):
         return pressure / (density ** self.gamma)
 
-    def build_field(self, t, coord=None):
+    def build_field(self, t, coord=None, keep_edge=False):
         """
         Populate the field data using the analytical solution.
 
         :param t:
         :param coord: If None, take the coordinate from the numerical solver.
         :return: None
         """
+
         if None is coord:
-            coord = self.svr.coord[::2]  # Use the numerical solver.
+            _ = self.svr.ncoord
```

Comment:
This 9 lines of code is mouthy.  Please use the follow code instead (please make proper adjustment if needed because I didn't test it myself):

```python
            _ = self.svr.ncoord
            start = 0 if keep_edge else 2
            stop = _ - stop - 1
            num =  # It makes more sense to calculate from start and stop
            xindices = np.linspace(start, stop, num=num, dtype='int32')
```